File: fleet_failure_ranking.m
function fleet_failure_ranking(machines)

num_machines = length(machines);

fprintf('\n=== Fleet Failure Risk Ranking ===\n')

failure_times = zeros(1,num_machines);

for m = 1:num_machines
failure_times(m) = machines(m).failure_time;
end

[sorted_times,idx] = sort(failure_times);

for i = 1:num_machines

machine_id = idx(i);

fprintf('Rank %d → Machine %d (Failure Time: %d)\n', ...
    i,machine_id,sorted_times(i));


end

File: fleet_failure_ranking.py
# ============================================================
# fleet_failure_ranking.py
# Fleet-level failure risk ranking
# ============================================================

import numpy as np


def fleet_failure_ranking(machines):

    num_machines  = len(machines)
    failure_times = np.zeros(num_machines)

    print("\n=== Fleet Failure Risk Ranking ===")

    for m in range(num_machines):
        ft = machines[m]['failure_time']
        failure_times[m] = ft if ft is not None else np.inf

    sorted_indices = np.argsort(failure_times)   # returns indices that would sort the array

    for rank, idx in enumerate(sorted_indices):
        machine_id = idx + 1   # display as 1-indexed for readability
        ft         = failure_times[idx]
        if ft == np.inf:
            print(f"Rank {rank+1} → Machine {machine_id} (No failure detected)")
        else:
            print(f"Rank {rank+1} → Machine {machine_id} (Failure Time: {int(ft)})")